### PR TITLE
Fix errors that abort the verification when debugging a DB

### LINKED ISSUE #2792: verifydb fails with debug in some cases
Using `./bin/instance verifydb -D` to check the `Data.fs` it fails in some cases:

```
Traceback (most recent call last):
  File "/Users/pbauer/.cache/buildout/eggs/Products.CMFPlone-5.2rc1-py3.7.egg/Products/CMFPlone/_scripts/verifydb.py", line 68, in verify_record
    class_info = unpickler.load()
  File "/Users/pbauer/.cache/buildout/eggs/ZODB-5.5.1-py3.7.egg/ZODB/_compat.py", line 62, in find_class
    return super(Unpickler, self).find_class(modulename, name)
ModuleNotFoundError: No module named 'Products.Archetypes'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./bin/instance", line 262, in <module>
    + sys.argv[1:]))
  File "/Users/pbauer/.cache/buildout/eggs/plone.recipe.zope2instance-6.1.3-py3.7.egg/plone/recipe/zope2instance/ctl.py", line 937, in main
    c.onecmd(' '.join(options.args))
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/cmd.py", line 217, in onecmd
    return func(arg)
  File "/Users/pbauer/.cache/buildout/eggs/Products.CMFPlone-5.2rc1-py3.7.egg/Products/CMFPlone/_scripts/verifydb.py", line 31, in zopectl_entry
    verify_zodb(app, debug=options.debug)
  File "/Users/pbauer/.cache/buildout/eggs/Products.CMFPlone-5.2rc1-py3.7.egg/Products/CMFPlone/_scripts/verifydb.py", line 50, in verify_zodb
    success = verify_record(oid, data, debug)
  File "/Users/pbauer/.cache/buildout/eggs/Products.CMFPlone-5.2rc1-py3.7.egg/Products/CMFPlone/_scripts/verifydb.py", line 82, in verify_record
    pickletools.dis(pickle[pos:])
UnboundLocalError: local variable 'pos' referenced before assignment
```

### PR DESCRIPTION
**Pull request by pbauer**
with `./bin/instance verifydb -D`
Fixes #2792

### CODE REVIEW

**Comment by pbauer** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/_scripts/verifydb.py [hunk 1]
```diff
@@ -77,11 +78,15 @@ def verify_record(oid, data, debug=False):
         ))
         logger.info(repr(pickle))
         logger.info(traceback.format_exc())
-        if debug:
+        if debug and pos is not None:
             try:
                 pickletools.dis(pickle[pos:])
+            except Exception:
+                logger.info(traceback.format_exc())
```
Comment (jensens) [inline Products/CMFPlone/_scripts/verifydb.py]: Use logger.exception("some message" ) here.

Comment (pbauer) [reply Products/CMFPlone/_scripts/verifydb.py]: Not loging them as exceptions is kind of the point here to allow continuing to walk through all the issues in the DB while having a pdb for each.

Comment (jensens) [reply Products/CMFPlone/_scripts/verifydb.py]: `logger.exception` does not raise, it just logs.

Comment (pbauer) [reply Products/CMFPlone/_scripts/verifydb.py]: Yes, but PDBDebugMode catches logger.exception which would be unwanted in thgis case.

Comment (jensens) [reply Products/CMFPlone/_scripts/verifydb.py]: so, I would add a comment in there, Otherwise your future self (or mine or others) dont now about it.

**Pull request merged**